pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -659,6 +659,40 @@ class Build {
         }
     }
 
+    def waitForANodeToBecomeActive(def label) {
+        def NodeHelper = context.library(identifier: 'openjdk-jenkins-helper@master').NodeHelper
+
+        if (NodeHelper.nodeIsOnline(label)) {
+            return
+        }
+
+        context.println("No active node matches this label: " + label)
+
+        int activeNodeTimeout = 0
+        if (${buildConfig.ACTIVE_NODE_TIMEOUT}.isInteger()) {
+            activeNodeTimeout = ${buildConfig.ACTIVE_NODE_TIMEOUT} as Integer
+        }
+
+
+        if (activeNodeTimeout > 0) {
+        context.println("Will check again periodically until either one comes online, or " + ${buildConfig.ACTIVE_NODE_TIMEOUT} + " minutes (ACTIVE_NODE_TIMEOUT) has passed.")
+            int x = 0
+            while (x < activeNodeTimeout) {
+                sleep(60 * 1000)
+                if (NodeHelper.nodeIsOnline(label)) {
+                    context.println("A node which matches this label is now active: " + label)
+                    return
```

Comment:
I'm not 100% sure that return statements are allowed in groovy while loops. Has this been tested to verify it works? If it doesn't, [break](https://www.tutorialspoint.com/groovy/groovy_break_statement.htm) should do the same thing

Reply:
Yes, but would it break out of the IF statement just outside that while loop? Because if not, we'd get an exception.

Reply:
And no, this hasn't been tested.

Reply:
It shouldn't do as `break` and `continue` statements usually only affect the loop that they have been declared inside. It shouldn't cascade and break the `if (activeNodeTimeout > 0) {` statement

Reply:
In which case replacing the return with a break would cause the exception I mentioned.

Reply:
I wrote some code to test this
```groovy
x = 15

def waitForANodeToBecomeActive() {
        def returnValue = ""
        while (x < 20) {
                println "X is currently " + x
                sleep(1000) // Sleep for 1 second

                if (x == 17) {
                    println("A node which matches this label is now active: " + "labelname")
                    returnValue = "RETURN BREAKS OUT OF THE FUNCTION"
                    return returnValue
                }

                x++
        }

        returnValue = "RETURN BREAKS OUT OF THE WHILE LOOP"
        return returnValue
}

def retValue = waitForANodeToBecomeActive()
println "RET VALUE = " + retValue
```
and it turns out I was wrong. A return statement inside a function will actually just break out of the function as you originally thought, it doesn't affect the while loop:
```bash
➜  ansible git:(master) ✗ groovy test.groovy
X is currently 15
X is currently 16
X is currently 17
A node which matches this label is now active: labelname
RET VALUE = RETURN BREAKS OUT OF THE FUNCTION
```
As such, I think this is a non issue assuming you want the code to return back to the function caller

Reply:
That is the intended behaviour, yes. Thanks for checking. :)